supabase/functions/auth/index.ts

```diff
@@ -0,0 +1,151 @@
+import { serve } from "https://deno.land/std/http/server.ts";
+import { createClient } from "https://esm.sh/@supabase/supabase-js";
+import { db } from "../_shared/db.ts";
+import { students } from "../_shared/schema.ts";
+import { getCookies, setCookie } from "https://deno.land/std/http/cookie.ts";
+
+type ValidationSchema = {
+  [key: string]: {
+    required?: boolean;
+    min?: number;
+    enum?: string[];
+  };
+};
+
+//need to move out the client create
+const supabase = createClient(
+  Deno.env.get("SUPABASE_URL")!,
+  Deno.env.get("SUPABASE_ANON_KEY")!,
+);
+
+const validate = (
+  body: Record<string, any>,
+  schema: ValidationSchema,
+): void => {
+  for (const key in schema) {
+    const rule = schema[key];
+    const value = body[key];
+
+    if (rule.required && (value === undefined || value === null)) {
+      throw new Error(`${key} is required.`);
+    }
+
+    if (rule.min && typeof value === "string" && value.length < rule.min) {
+      throw new Error(`${key} must be at least ${rule.min} characters long.`);
+    }
+
+    if (rule.enum && !rule.enum.includes(value)) {
+      throw new Error(`${key} must be one of: ${rule.enum.join(", ")}.`);
+    }
+  }
+};
+
+serve(async (req) => {
```

Comment:
Migrate the entire function code to `HonoJS`. Currently, the code is quite smelly for the function. Would be cleaner to use the framework and simplify route structure. Gives you ExpressJS like powers.